File: app/src/main/java/com/sayler/gina/dagger/component/ActivityComponent.java
package com.sayler.gina.dagger.component;

import android.app.Activity;
import com.sayler.gina.dagger.module.ActivityModule;
import dagger.Component;

@Component(
    modules = {
        ActivityModule.class
    },
    dependencies = {
        ApplicationComponent.class
    }
)
public interface ActivityComponent {

  // Provide
  Activity activity();
}

File: app/src/main/java/com/sayler/gina/dagger/component/ActivityComponent.kt
package com.sayler.gina.dagger.component

import android.app.Activity
import com.sayler.gina.dagger.module.ActivityModule
import dagger.Component

@Component(modules = arrayOf(ActivityModule::class), dependencies = arrayOf(ApplicationComponent::class))
interface ActivityComponent {

    // Provide
    fun activity(): Activity
}